.github/workflows/ci.yml

```diff
@@ -0,0 +1,62 @@
+name: Build
+
+on:
+  pull_request:
+    branches: ['**']
+  push:
+    branches: ['**']
+    tags: [v*]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        temurin_tag:
+          - 8u322-b06-jdk
+          - 11.0.14.1_1-jdk
+        architecture:
+          - linux/amd64
+          - linux/arm/v8
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # https://github.com/docker/setup-qemu-action
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Docker Hub
+        if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/v'))
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      # https://github.com/docker/metadata-action
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          # list of Docker images to use as base name for tags
+          images: |
+            dwolla/jenkins-agent-core
+          # Docker tags based on the following events/attributes
+          tags: |
+            type=sha
+
+      # https://github.com/docker/build-push-action
+      - name: Build and push
```

Comment:
Probably more like a nice to have, but have you looked into [using](https://developer.ibm.com/tutorials/createmulti-architecture-docker-images/) `docker manifest` to have docker automatically use the correct image based on the client's architecture?

Reply:
It doesn't look like you can build multi-arch images locally and have them stay in your local images—AFAICT you have to push them to a registry (or leave them in the `buildx` cache, but they can't really be used for anything there):

```
$ docker buildx build \
  --build-arg TEMURIN_TAG=8u322-b06-jdk \
  --label org.opencontainers.image.title=jenkins-agent-docker-core \
  --label "org.opencontainers.image.description=Docker image containing the Jenkins remoting JAR and required dependencies" \
  --label org.opencontainers.image.url=https://github.com/Dwolla/jenkins-agent-docker-core \
  --label org.opencontainers.image.source=https://github.com/Dwolla/jenkins-agent-docker-core \
  --label org.opencontainers.image.version=sha-91a4884 \
  --label org.opencontainers.image.created=2022-03-21T22:24:37.318Z \
  --label org.opencontainers.image.revision=91a4884d8c90f348b059629216fc2c6baf0f7b5e \
  --label org.opencontainers.image.licenses=MIT \
  --platform linux/amd64,linux/arm/v8 \
  --tag dwolla/jenkins-agent-core:sha-91a4884 \
  --output=type=docker \
  .
[+] Building 0.0s (0/0)                                                                                                                                         
error: docker exporter does not currently support exporting manifest lists
```

If I try omitting `--output=type=docker` then it looks like it does successfully build both platforms, though. And if I do them one at a time with `--output=type=docker` in the command, they both show up in `docker images`—but the second one ends up overwriting the first one.

I'm a little unclear whether this means we need to build all architectures for a single tag in a single `docker buildx build` command, or whether we can build them separately and push them to the same place. Before `docker buildx` was a thing, I think the way you did this was just to build them on separate machines and then push them to the same tag, so that gives me hope that the current approach of building each architecture in a separate build matrix run will work. But if it doesn't work, we can collapse that axis of the GHA build matrix in a followup PR.